src/efi/mod.rs

```diff
@@ -1031,47 +1157,106 @@ struct LoadedImageWrapper {
     entry_point: u64,
 }
 
-type DevicePaths = [file::FileDevicePathProtocol; 2];
+#[repr(C)]
+struct FileDevicePathProtocol {
+    pub device_path: DevicePathProtocol,
+    pub filename: [u16; 128],
```

Comment:
3c75868fb342ec7763bf94ddd6999afa292d9f8a increases the max path length to 256, but the size of this field remains at 128 length. It should be 256 because UEFI uses a character in UCS2 that represents each character in 16-bits.
```suggestion
    pub filename: [u16; 256],
```